library/ssl_tls.c

```diff
@@ -4888,81 +4975,107 @@ void mbedtls_ssl_optimize_checksum( mbedtls_ssl_context *ssl,
     defined(MBEDTLS_SSL_PROTO_TLS1_1)
-     mbedtls_md5_update_ret( &ssl->handshake->fin_md5 , buf, len );
-    mbedtls_sha1_update_ret( &ssl->handshake->fin_sha1, buf, len );
+    if( ( ret = mbedtls_md5_update_ret( &ssl->handshake->fin_md5 , buf, len ) ) != 0 )
+        return ( ret );
+    if( ( ret = mbedtls_sha1_update_ret( &ssl->handshake->fin_sha1, buf, len ) ) != 0 )
+        return ( ret );
 #endif
 #if defined(MBEDTLS_SSL_PROTO_TLS1_2)
 #if defined(MBEDTLS_SHA256_C)
-    mbedtls_sha256_update_ret( &ssl->handshake->fin_sha256, buf, len );
+    if( ( ret = mbedtls_sha256_update_ret( &ssl->handshake->fin_sha256, buf, len ) ) != 0 )
+        return ( ret );
 #endif
 #if defined(MBEDTLS_SHA512_C)
-    mbedtls_sha512_update_ret( &ssl->handshake->fin_sha512, buf, len );
+if( ( ret = mbedtls_sha512_update_ret( &ssl->handshake->fin_sha512, buf, len ) ) != 0 )
+        return ( ret );
 #endif
 #endif /* MBEDTLS_SSL_PROTO_TLS1_2 */
+    return ( ret );
 }
 
 #if defined(MBEDTLS_SSL_PROTO_SSL3) || defined(MBEDTLS_SSL_PROTO_TLS1) || \
     defined(MBEDTLS_SSL_PROTO_TLS1_1)
-static void ssl_update_checksum_md5sha1( mbedtls_ssl_context *ssl,
+static int ssl_update_checksum_md5sha1( mbedtls_ssl_context *ssl,
                                          const unsigned char *buf, size_t len )
 {
-     mbedtls_md5_update_ret( &ssl->handshake->fin_md5 , buf, len );
-    mbedtls_sha1_update_ret( &ssl->handshake->fin_sha1, buf, len );
+    int ret;
+    if( ( ret = mbedtls_md5_update_ret( &ssl->handshake->fin_md5 , buf, len ) ) != 0 )
+        return ( ret );
+    if( ( ret = mbedtls_sha1_update_ret( &ssl->handshake->fin_sha1, buf, len ) ) != 0 )
+        return ( ret );
+
+    return ( ret );
 }
 #endif
 
 #if defined(MBEDTLS_SSL_PROTO_TLS1_2)
 #if defined(MBEDTLS_SHA256_C)
-static void ssl_update_checksum_sha256( mbedtls_ssl_context *ssl,
+static int ssl_update_checksum_sha256( mbedtls_ssl_context *ssl,
                                         const unsigned char *buf, size_t len )
 {
-    mbedtls_sha256_update_ret( &ssl->handshake->fin_sha256, buf, len );
+    int ret;
+
+    ret = mbedtls_sha256_update_ret( &ssl->handshake->fin_sha256, buf, len );
+
+    return ( ret );
 }
 #endif
 
 #if defined(MBEDTLS_SHA512_C)
-static void ssl_update_checksum_sha384( mbedtls_ssl_context *ssl,
+static int ssl_update_checksum_sha384( mbedtls_ssl_context *ssl,
                                         const unsigned char *buf, size_t len )
 {
-    mbedtls_sha512_update_ret( &ssl->handshake->fin_sha512, buf, len );
+    int ret;
+
+    ret = mbedtls_sha512_update_ret( &ssl->handshake->fin_sha512, buf, len );
+
+    return ( ret );
 }
 #endif
 #endif /* MBEDTLS_SSL_PROTO_TLS1_2 */
 
 #if defined(MBEDTLS_SSL_PROTO_SSL3)
-static void ssl_calc_finished_ssl(
+static int ssl_calc_finished_ssl(
                 mbedtls_ssl_context *ssl, unsigned char *buf, int from )
 {
     const char *sender;
     mbedtls_md5_context  md5;
     mbedtls_sha1_context sha1;
-
```

Comment:
I don't think we should make changes to the code outside the PR's scope. I would reintroduce this empty line.